src/main/scala/com/raphtory/core/components/graphbuilder/GraphBuilder.scala

```diff
@@ -9,6 +9,69 @@ import com.typesafe.config.Config
 import scala.collection.mutable
 import scala.collection.mutable.ArrayBuffer
 
+/**
+  * {s}`GraphBuilder`
+  *
+  * : `GraphBuilder` trait for creating a Graph by adding and deleting vertices and edges.
+  *    While using `GraphBuilder`, override {s}`parseTuple(tuple: T)` to define parsing of rows and use methods {s}`addVertex`, {s}`addEdge`
+  *    for adding vertices and edge to the graph. Deletion can be performed with {s}`deleteEdge` and {s}`deleteVertex` methods.
+  *
+  * ## Methods
+  *
+  *    {s}`addVertex(updateTime: Long, srcId: Long)`
+  *      : Adds vertex for a timestamp {s}`updateTime` and ID {s}`srcId`
+  *
+  *    {s}`addVertex(updateTime: Long, srcId: Long, properties: Properties)`
+  *      : Adds vertex for a timestamp {s}`updateTime` and ID {s}`srcId`, with Properties {s}`properties`
+  *
+  *    {s}`addVertex(updateTime: Long, srcId: Long, vertexType: Type))`
+  *      : Adds vertex for a timestamp {s}`updateTime` and ID {s}`srcId` for vertex type {s}`vertexType`
+  *
+  *    {s}`addVertex(updateTime: Long, srcId: Long, properties: Properties, vertexType: Type)`
+  *      : Adds vertex for timestamp {s}`updateTime`, ID {s}`srcId`, Properties {s}`properties`, vertex type {s}`vertexType`
+  *
+  *    {s}`deleteVertex(updateTime: Long, srcId: Long)`
+  *      : Deletes vertex for timestamp {s}`updateTime`, ID {s}`srcId`
```

Comment:
what does a delete vertex do? its doesnt delete the actual vertex right? it creates a new " vertex delete" object  that a specific time, removes the vertex from the graph scope, the vertex is still available in previous scopes, this object is sent as an update?